crypto/evp/asymcipher.c

```diff
@@ -159,18 +160,22 @@ static int evp_pkey_asym_cipher_init(EVP_PKEY_CTX *ctx, int operation,
         goto err;
     }
 
+    desc = cipher->description != NULL ? cipher->description
+        : "no asym cipher description by provider";
     switch (operation) {
     case EVP_PKEY_OP_ENCRYPT:
         if (cipher->encrypt_init == NULL) {
-            ERR_raise(ERR_LIB_EVP, EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
+            ERR_raise_data(ERR_LIB_EVP, EVP_R_PROVIDER_ASYM_CIPHER_MISSING,
```

Comment:
Why do you change the reason code? IMO the original one was better.  If you want you could put `asym_cipher_encrypt_init` into the text below to make it clear what operation is not provided.

Reply:
I've removed all the tentative new `EVP_R_PROVIDER_*_MISSING` reason codes
and now use in all such cases `EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE`.